### PR TITLE
Consider passwordless signins when provisioning accounts

### LINKED ISSUE #5933: Passwordless users cannot provision accounts
Attempting to provision accounts (either via first visit to `apps/mail` where the user gets a 500 error, or via mass provisioning) chokes when trying to set password [here](https://github.com/nextcloud/mail/blob/d9f3cc1de93d70368fed5fe6613a6ca0043e59cc/lib/Http/Middleware/ProvisioningMiddleware.php#L75).

`PUT /apps/mail/api/settings/provisioning/all`
```
TypeError: OCA\Mail\Service\Provisioning\Manager::updatePassword(): Argument #2 ($password) must be of type string, null given, called in /var/lib/nextcloud/apps/mail/lib/Http/Middleware/ProvisioningMiddleware.php on line 75  in file '/var/lib/nextcloud/apps/mail/lib/Service/Provisioning/Manager.php' line 300

/var/lib/nextcloud/apps/mail/lib/Http/Middleware/ProvisioningMiddleware.php - line 75:
OCA\Mail\Service\Provisioning\Manager->updatePassword("*** sensiti ... *")

/usr/share/webapps/nextcloud/lib/private/AppFramework/Middleware/MiddlewareDispatcher.php - line 97:
OCA\Mail\Http\Middleware\ProvisioningMiddleware->beforeController()

/usr/share/webapps/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 118:
OC\AppFramework\Middleware\MiddlewareDispatcher->beforeController()

/usr/share/webapps/nextcloud/lib/private/AppFramework/App.php - line 157:
OC\AppFramework\Http\Dispatcher->dispatch()

/usr/share/webapps/nextcloud/lib/private/Route/Router.php - line 302:
OC\AppFramework\App::main()

/usr/share/webapps/nextcloud/lib/base.php - line 1006:
OC\Route\Router->match()

/usr/share/webapps/nextcloud/index.php - line 36:
OC::handleRequest()
```

- Nextcloud Hub 23.0.0
- Alpine Linux edge (5.15.13-0-lts)
- PHP 8.0.14 (built: Dec 17 2021 20:46:56) ( NTS )
- mail: 1.11.5

I do not know if an issue should be opened under `server` or `user_saml` for this, or if any other backends have this issue.

### PR DESCRIPTION
**Pull request by miaulalala**
Add a null check on the password and skip updating the password for these accounts.

Fixes #5933

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

File: lib/Http/Middleware/ProvisioningMiddleware.php [hunk 1]
```diff
@@ -70,9 +70,15 @@ public function beforeController($controller, $methodName) {
 		}
 		try {
 			$this->provisioningManager->provisionSingleUser($configs, $user);
+			$password = $this->credentialStore->getLoginCredentials()->getPassword();
```
Comment (ChristophWurst) [inline lib/Http/Middleware/ProvisioningMiddleware.php:73]: huh, so `\OCP\Authentication\LoginCredentials\ICredentials::getPassword` lies? Time to fix that in server to specify a nullable string

Comment (ChristophWurst) [reply lib/Http/Middleware/ProvisioningMiddleware.php:73]: @miaulalala could you please take care of this?

Comment (miaulalala) [reply lib/Http/Middleware/ProvisioningMiddleware.php:73]: Done here https://github.com/nextcloud/server/pull/31377/files

**Review by ChristophWurst — APPROVED** [review]
Provisioned accounts might still not work if the user never logs into a session with password. But an occasional passwordless login will now be possible :+1:

**Comment by MohammedNoureldin** [comment]
I am not sure what the current status of this is. I am trying to use a mailserver that uses OAuth2 for authentication, but I still see no connections coming at all after provisioning the mail or when trying to access the mail app.

Does NextCloud forward the tokens or try to connect at all with the mail server for users logged in using OIDC?

**Pull request merged**